packages/indexer-client/src/IndexerBaseClient.ts

```diff
@@ -1087,6 +1093,44 @@ export class IndexerBaseClient {
     return response.data as GetIndexerVrtxTokenInfoResponse;
   }
 
+  /**
+   * Retrieves staking top stakers.
+   *
+   * @param params
+   */
+  async getStakingV2TopStakers(
+    params: GetIndexerStakingV2TopStakersParams,
+  ): Promise<GetIndexerStakingV2TopStakersResponse> {
+    const baseResponse = await this.query('staking_v2_top_stakers', {
+      limit: params.limit,
+    });
+
+    return {
+      stakers: baseResponse.stakers.map(mapIndexerStakingV2Staker),
+    };
+  }
+
+  /**
+   * Retrieves staking v2 pool snapshots.
+   *
+   * @param params
+   */
+  async getStakingV2PoolSnapshots(
+    params: GetIndexerStakingV2PoolSnapshotsParams,
+  ): Promise<GetIndexerStakingV2PoolSnapshotsResponse> {
+    const baseResponse = await this.query('staking_v2_pool_snapshots', {
+      interval: {
+        count: params.limit,
+        max_time: params.maxTimeInclusive?.toString(),
```

Comment:
nit: would a `toFixed` be a better conversion here?

Reply:
Will switch all to toFixed().

Reply:
Should this be `toFixed(0)`?

Reply:
I don't think there is a difference.

Reply:
On mobile so I cannot check, but if this is a BigDecimal then there is: https://mikemcl.github.io/bignumber.js/#toFix

Reply:
https://github.com/vertex-protocol/vertex-typescript-sdk/pull/280/files